driver/src/test/java/org/neo4j/driver/v1/EncryptionTest.java

```diff
@@ -0,0 +1,192 @@
+        assertThat( e.getMessage(), startsWith( "Connection to the database terminated" ) );
+    }
+
+    private static Config newConfig( boolean withEncryption )
+    {
+        return withEncryption ? configWithEncryption() : configWithoutEncryption();
+    }
+
+    private static Config configWithEncryption()
+    {
+        return Config.build().withEncryption().toConfig();
+    }
+
+    private static Config configWithoutEncryption()
+    {
+        return Config.build().withoutEncryption().toConfig();
+    }
+
+    enum ClientSocketStatus
+    {
+        Encrypted,
+        Unencrypted,
+        Unknown,
+        FailedToRecognize
+    }
+
+    public final class DummyNettyServer
+    {
+        private EventLoopGroup bossGroup;
+        private EventLoopGroup workerGroup;
+        private Channel serverChannel;
+        private final AtomicReference<ClientSocketStatus> clientStatus = new AtomicReference<>( ClientSocketStatus.Unknown );
+
+        public int start() throws Throwable
+        {
+            bossGroup = new NioEventLoopGroup( 1 );
+            workerGroup = new NioEventLoopGroup();
+
+            ServerBootstrap server = new ServerBootstrap();
+            server.group( bossGroup, workerGroup ).channel( NioServerSocketChannel.class )
+                    .childHandler( new ChannelInitializer<SocketChannel>()
+            {
+                @Override
+                public void initChannel( SocketChannel ch ) throws Exception
+                {
+                    ChannelPipeline p = ch.pipeline();
+                    p.addLast( new ByteToMessageDecoder()
+                    {
+                        @Override
+                        protected void decode( ChannelHandlerContext ctx, ByteBuf in, List<Object> list ) throws Exception
+                        {
+                            // Will use the first five bytes to detect a protocol.
+                            if ( in.readableBytes() < 5 )
+                            {
+                                return;
+                            }
+
+                            checkClientSocketStatus( in );
+                            ctx.close();
+                        }
+
+                        @Override
+                        public void exceptionCaught( ChannelHandlerContext ctx, Throwable cause )
+                        {
+                            // Close the connection when an exception is raised.
+                            cause.printStackTrace();
+                            ctx.close();
+                        }
+                    } );
+                }
+            } );
+            // Start the server.
+            serverChannel = server.bind( 0 ).sync().channel();
+            return ((InetSocketAddress) serverChannel.localAddress()).getPort();
+        }
+
+        public void stop() throws Throwable
+        {
+            try
+            {
+                // Now shutting down
+                serverChannel.close().sync();
```

Comment:
I think it should be enough to just shutdown the event loop group. Channel gets terminated as part of the event loop group termination. `serverChannel` field might not be needed in this case.